07-errors.md

```diff
@@ -0,0 +1,298 @@
+| `ErrNotFound`        | target actor doesn't exist                                |
+| `ErrIllegalArgument` | if the passed address buffer isn't valid, in memory, etc. |
+
+#### `actor::new_actor_address`
+
+TODO (likely needs to go)
+
+#### `actor::create_actor`
+
+TODO
+
+#### `crypto::verify_signature`
+
+| Error                | Reason                                                |
+|----------------------|-------------------------------------------------------|
+| `ErrIllegalArgument` | signature, address, or plaintext buffers are invalid. |
+
+#### `crypto::hash_blake2b`
+
+| Error                | Reason         |
+|----------------------|----------------|
+| `ErrIllegalArgument` | invalid buffer |
+
+#### `crypto::verify_seal`
+
+| Error                | Reason   |
+|----------------------|----------|
+| `ErrIllegalArgument` | anything |
+
+#### `crypto::verify_post`
+
+| Error                | Reason   |
+|----------------------|----------|
+| `ErrIllegalArgument` | anything |
+
+#### `crypto::compute_unsealed_sector_cid`
+
+| Error                | Reason          |
+|----------------------|-----------------|
+| `ErrIllegalArgument` | everything else |
+
+#### `crypto::verify_consensus_fault`
+
+| Error                | Reason                                 |
+|----------------------|----------------------------------------|
+| `ErrLimit`           | exceeded lookback limit finding block. |
+| `ErrIllegalArgument` | something else                         |
+
+#### `crypto::verify_aggregate_seals`
+
+| Error                | Reason                          |
+|----------------------|---------------------------------|
+| `ErrLimit`           | exceeds seal aggregation limit. |
+| `ErrIllegalArgument` | something is malformed          |
+
+#### `crypto::batch_verify_seals`
+
+| Error                | Reason              |
+|----------------------|---------------------|
+| `ErrIllegalArgument` | if malformed params |
+
+#### `rand::get_*_randomness`
+
+| Error                | Reason                  |
+|----------------------|-------------------------|
+| `ErrLimit`           | lookback exceeds limit. |
+| `ErrIllegalArgument` | invalid buffer, etc.    |
+
+#### `gas::charge_gas`
+
+| Error                | Reason               |
+|----------------------|----------------------|
+| `ErrIllegalArgument` | invalid name buffer. |
+
+#### `send::send`
+
+| Error                  | Reason                                               |
+|------------------------|------------------------------------------------------|
+| `ErrNotFound`          | target actor does not exist and cannot be created.   |
+| `ErrInsufficientFunds` | tried to send more FIL than available.               |
+| `ErrInvalidHandle`     | parameters block not found.                          |
+| `ErrLimit`             | recursion limit reached.                             |
+| `ErrIllegalActor`      | called actor panicked or returned an invalid result. |
```

Comment:
This error is a bit strange and I'm not convinced it's correct. Usually, a syscall error means the _caller_ did something wrong. In this case, it means the _callee_ failed.

I'm thinking we should use a system abnormal exit code here instead (e.g., the actor panic sys exit code).

cc @raulk